src/bind.test.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+const bind = require('./bind');
+
+describe('Basic scenario', () => {
+  const f = (...args) => args;
+  let device;
+  const argsBase = [1, 2, 3];
+  const argsExtra = [4, 5, 6];
+
+  beforeEach(() => {
+    device = bind(f, ...argsBase);
+  });
+
+  test('should return expected array', () => {
+    expect(device())
+      .toStrictEqual([1, 2, 3]);
+  });
+
+  test('should return expected array', () => {
+    expect(device(...argsExtra))
+      .toStrictEqual([1, 2, 3, 4, 5, 6]);
+  });
+
+  test('should not change base and extra params', () => {
+    expect(argsBase)
+      .toStrictEqual([1, 2, 3]);
+    expect(argsExtra)
+      .toStrictEqual([4, 5, 6]);
```

Comment:
use pass the params as ...
so it is not possible to change them
so this code checks nothing